clippy_lints/src/copies.rs

```diff
@@ -323,20 +325,32 @@ where
     None
 }
 
-fn search_same<T, Hash, Eq>(exprs: &[T], hash: Hash, eq: Eq) -> Option<(&T, &T)>
+fn search_common_cases<'a, T, Eq>(exprs: &'a [T], eq: &Eq) -> Option<(&'a T, &'a T)>
 where
-    Hash: Fn(&T) -> u64,
     Eq: Fn(&T, &T) -> bool,
 {
-    // common cases
     if exprs.len() < 2 {
-        return None;
+        None
     } else if exprs.len() == 2 {
-        return if eq(&exprs[0], &exprs[1]) {
+        if eq(&exprs[0], &exprs[1]) {
             Some((&exprs[0], &exprs[1]))
         } else {
             None
-        };
+        }
+    } else {
+        None
+    }
+}
+
+fn search_same<T, Hash, Eq>(exprs: &[T], hash: Hash, eq: Eq) -> Option<(&T, &T)>
```

Comment:
Do we even need this function anymore? Now it is used in the if-part of this lint.
Shouldn't search_same_List work there as well?

Reply:
I did not want to refactor too much right away.

Reply:
Got it! Do you want to add this to this commit later? Or maybe we could open an issue for this as well.

Reply:
I am doing it right now, just checking again that everything works as intended.

Reply:
Tests pass without changes in the if stderr.

Reply:
Cool!
Although that just means, that there isn't a change in there that has multiple identical branches?